models/fivetran_platform__connector_status.sql

```diff
@@ -177,16 +187,27 @@ final as (
         coalesce(schema_changes.number_of_schema_changes_last_month, 0) as number_of_schema_changes_last_month
         
         {% if var('fivetran_platform_using_sync_alert_messages', true) %}
-        , {{ fivetran_utils.string_agg("distinct case when connector_recent_logs.event_type = 'SEVERE' then connector_recent_logs.message_data else null end", "'\\n'") }} as errors_since_last_completed_sync
-        , {{ fivetran_utils.string_agg("distinct case when connector_recent_logs.event_type = 'WARNING' then connector_recent_logs.message_data else null end", "'\\n'") }} as warnings_since_last_completed_sync
+        , {{ fivetran_utils.string_agg("case when connector_recent_logs.event_type = 'SEVERE' then connector_recent_logs.message_data else null end", "'\\n'") }} as errors_since_last_completed_sync
+        , {{ fivetran_utils.string_agg("case when connector_recent_logs.event_type = 'WARNING' then connector_recent_logs.message_data else null end", "'\\n'") }} as warnings_since_last_completed_sync
```

Comment:
sqlserver doesn't support `distinct` with string/list_agg -- i was wondering though if we should include the `distinct` for non sqlserver targets 

i do think that the `group bys` we have in the previous and current CTEs should also avoid duplicate messages however

Reply:
Hmm actually I don't know if the `group by`s will remove the duplicates in the string_agg. I worry we can still end up with a list of duplicates within the string agg'd field.

Actually, hasn't this logic been a problem for some users due to the length of the sring_agg (particularly for redshift users)? What are your thoughts on us taking a more cautious approach and just not using this code for sql server users? I do think removing the `distinct` for all other warehouses could lead to unforeseen issues, and I worry adding extra conditionals to these fields will unnecessarily add more code to maintain. 

I can also see these fields giving users issues when running on production sql server data. Especially because apparently the `string_agg` function has a [limit of 8000 characters](https://www.mssqltips.com/sqlservertutorial/9371/sql-string-agg-function/#:~:text=By%20default%2C%20the%20STRING_AGG%20is,a%20limit%20of%208000%20characters.).

Thoughts?

Reply:
updated to avoid on sqlserver